From b4c0fc7219af0580f85d557eb4628a982af95f2b Mon Sep 17 00:00:00 2001
From: Jian-Hong Pan <jhp@endlessos.org>
Date: Tue, 15 Apr 2025 17:03:25 +0800
Subject: [PATCH 2/3] [Endless] linux-pam: Enable usergroups for private user
 groups, the UMASK 0002

Enable usergroups for private user groups, the UMASK 0002 by following
Ubuntu settings.

https://phabricator.endlessm.com/T31847
https://phabricator.endlessm.com/T35802
---
 elements/components/linux-pam-base.bst | 1 +
 files/linux-pam-config/password-auth   | 5 +++++
 files/linux-pam-config/system-auth     | 5 +++++
 3 files changed, 11 insertions(+)

diff --git a/elements/components/linux-pam-base.bst b/elements/components/linux-pam-base.bst
index 0aa51500d..1148d37c4 100644
--- a/elements/components/linux-pam-base.bst
+++ b/elements/components/linux-pam-base.bst
@@ -17,6 +17,7 @@ variables:
     -Dnis=disabled
     -Dselinux=disabled
     -Ddocs=disabled
+    -Dusergroups=true
 
 config:
   install-commands:
diff --git a/files/linux-pam-config/password-auth b/files/linux-pam-config/password-auth
index 1df8aac70..0e9e0f2a5 100644
--- a/files/linux-pam-config/password-auth
+++ b/files/linux-pam-config/password-auth
@@ -15,4 +15,9 @@ session     optional      pam_keyinit.so revoke
 session     required      pam_limits.so
 -session     optional      pam_systemd.so
 session     [success=1 default=ignore] pam_succeed_if.so service in crond quiet use_uid
+# The pam_umask module will set the umask according to the system default in
+# /etc/login.defs and user settings, solving the problem of different
+# umask settings with different shells, display managers, remote sessions etc.
+# See "man pam_umask".
+session     optional      pam_umask.so
 session     required      pam_unix.so
diff --git a/files/linux-pam-config/system-auth b/files/linux-pam-config/system-auth
index 1df8aac70..0e9e0f2a5 100644
--- a/files/linux-pam-config/system-auth
+++ b/files/linux-pam-config/system-auth
@@ -15,4 +15,9 @@ session     optional      pam_keyinit.so revoke
 session     required      pam_limits.so
 -session     optional      pam_systemd.so
 session     [success=1 default=ignore] pam_succeed_if.so service in crond quiet use_uid
+# The pam_umask module will set the umask according to the system default in
+# /etc/login.defs and user settings, solving the problem of different
+# umask settings with different shells, display managers, remote sessions etc.
+# See "man pam_umask".
+session     optional      pam_umask.so
 session     required      pam_unix.so
-- 
2.49.0

